pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -26,72 +26,94 @@ import (
+	pvcStorageClassDeviceSetPVCId, pvcStorageClassDeviceSetPVCIdLabelSelector = makeStorageClassDeviceSetPVCIDNew(storageClassDeviceSetName, pvcTemplate.GetName(), setIndex)
+	pvc = makeStorageClassDeviceSetPVC(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId, setIndex, pvcTemplate)
 	presentPVCs, err := c.context.Clientset.CoreV1().PersistentVolumeClaims(c.Namespace).List(metav1.ListOptions{LabelSelector: pvcStorageClassDeviceSetPVCIdLabelSelector})
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to create pvc %s for storageClassDeviceSet %s", pvc.GetGenerateName(), storageClassDeviceSet.Name)
+		return nil, errors.Wrapf(err, "failed to create pvc %s for storageClassDeviceSet %s", pvc.GetGenerateName(), storageClassDeviceSetName)
 	}
-	if len(presentPVCs.Items) == 0 { // No PVC found, creating a new one
+
+	presentPVCsNum := len(presentPVCs.Items)
+	// No PVC found, creating a new one
+	if presentPVCsNum == 0 {
 		deployedPVC, err := c.context.Clientset.CoreV1().PersistentVolumeClaims(c.Namespace).Create(pvc)
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to create pvc %s for storageClassDeviceSet %s", pvc.GetGenerateName(), storageClassDeviceSet.Name)
+			return nil, errors.Wrapf(err, "failed to create pvc %q for storageClassDeviceSet %q", pvc.GetGenerateName(), storageClassDeviceSetName)
 		}
+		logger.Debugf("just created pvc %q", deployedPVC.Name)
 		return deployedPVC, nil
-	} else if len(presentPVCs.Items) == 1 { // The PVC is already present.
+		// The PVC is already present
+	} else if presentPVCsNum == 1 {
+		logger.Debugf("already present pvc %q", presentPVCs.Items[0].Name)
+		// Updating with the new label
 		return &presentPVCs.Items[0], nil
 	}
 	// More than one PVC exists with same labelSelector
-	return nil, errors.Errorf("more than one PVCs exists with label %s, pvcs %s", pvcStorageClassDeviceSetPVCIdLabelSelector, presentPVCs)
+	return nil, errors.Errorf("more than one PVCs exists with label %q, pvcs %q", pvcStorageClassDeviceSetPVCIdLabelSelector, presentPVCs)
 }
 
-func makeStorageClassDeviceSetPVC(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId string, pvcIndex, setIndex int, pvcTemplate v1.PersistentVolumeClaim) (pvcs *v1.PersistentVolumeClaim) {
-	pvcLabels := makeStorageClassDeviceSetPVCLabel(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId, pvcIndex, setIndex)
+func makeStorageClassDeviceSetPVC(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId string, setIndex int, pvcTemplate v1.PersistentVolumeClaim) (pvcs *v1.PersistentVolumeClaim) {
+	pvcLabels := makeStorageClassDeviceSetPVCLabel(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId, setIndex)
 
 	// pvc naming format <storageClassDeviceSetName>-<SetNumber>-<PVCIndex>
 	pvcGenerateName := pvcStorageClassDeviceSetPVCId + "-"
 
-	// Add pvcTemplate name to pvc name. i.e <storageClassDeviceSetName>-<SetNumber>-<PVCIndex>-<pvcTemplateName>
-	if len(pvcTemplate.GetName()) != 0 {
```

Comment:
The data and metadata PVC names need to be different. If this is removed it seems the names would be the same. Do you have an example of the PVC names that were generated based on the deviceSet to confirm they are still unique?

Reply:
I'm removing this because it's now including as part of `makeStorageClassDeviceSetPVCIDNew()`.

```
persistentvolumeclaim "set1-data-0-w4bgt" deleted
persistentvolumeclaim "set1-data-1-8c577" deleted
persistentvolumeclaim "set1-data-2-q4c6p" deleted
persistentvolumeclaim "set1-metadata-0-wt5wt" deleted
persistentvolumeclaim "set1-metadata-1-7vz6c" deleted
persistentvolumeclaim "set1-metadata-2-kppl8" deleted
```